mysql-test/suite/sys_vars/t/innodb_default_encryption_key_id_basic.test

```diff
@@ -1,3 +1,108 @@
+SELECT @@global.innodb_default_encryption_key_id;
+
+--let $error_log=$MYSQL_TMP_DIR/my_restart.err
+
+--replace_result $KEYRING_PLUGIN KEYRING_PLUGIN $MYSQL_TMP_DIR MYSQL_TMP_DIR
+--let $restart_parameters=restart:--early-plugin-load="keyring_file=$KEYRING_PLUGIN" --keyring_file_data=$MYSQL_TMP_DIR/keyring/keyring --default-table-encryption=ONLINE_TO_KEYRING --innodb-encryption-rotate-key-age=15 --innodb-encryption-threads=4 --innodb-default-encryption-key-id=4294967295 --log-error=$error_log
+--replace_result $MYSQL_TMP_DIR MYSQL_TMP_DIR
+--source include/restart_mysqld.inc
+
+--let ABORT_ON=NOT_FOUND
+--let SEARCH_FILE=$error_log
+--let SEARCH_PATTERN=\[Server\] option 'innodb-default-encryption-key-id': unsigned value 4294967295 adjusted to 4294967294\.
+--source include/search_pattern_in_file.inc
+
+--let $restart_parameters=restart:--early-plugin-load="keyring_file=$KEYRING_PLUGIN" --keyring_file_data=$MYSQL_TMP_DIR/keyring/keyring --default-table-encryption=ONLINE_TO_KEYRING --innodb-encryption-rotate-key-age=15 --innodb-encryption-threads=4 --innodb-default-encryption-key-id=-1 --log-error=$error_log
+--replace_result $MYSQL_TMP_DIR MYSQL_TMP_DIR
+--source include/restart_mysqld.inc
+
+--let ABORT_ON=NOT_FOUND
+--let SEARCH_FILE=$error_log
+--let SEARCH_PATTERN=\[Server\] option 'innodb-default-encryption-key-id': value -1 adjusted to 0\.
+--source include/search_pattern_in_file.inc
+
+SET innodb_default_encryption_key_id=4294967295;
+SELECT @@session.innodb_default_encryption_key_id;
+SELECT @@global.innodb_default_encryption_key_id;
+
+SET global innodb_default_encryption_key_id=4294967295;
+SELECT @@session.innodb_default_encryption_key_id;
+SELECT @@global.innodb_default_encryption_key_id;
+
+SET persist innodb_default_encryption_key_id=4294967295;
+SELECT @@session.innodb_default_encryption_key_id;
+SELECT @@global.innodb_default_encryption_key_id;
+
+SET persist_only innodb_default_encryption_key_id=4294967295;
+SELECT @@session.innodb_default_encryption_key_id;
+SELECT @@global.innodb_default_encryption_key_id;
+
+SET innodb_default_encryption_key_id=-1;
+SELECT @@session.innodb_default_encryption_key_id;
+SELECT @@global.innodb_default_encryption_key_id;
+
+SET global innodb_default_encryption_key_id=-1;
+SELECT @@session.innodb_default_encryption_key_id;
+SELECT @@global.innodb_default_encryption_key_id;
+
+SET persist innodb_default_encryption_key_id=-1;
+SELECT @@session.innodb_default_encryption_key_id;
+SELECT @@global.innodb_default_encryption_key_id;
 
+SET persist_only innodb_default_encryption_key_id=-1;
+SELECT @@session.innodb_default_encryption_key_id;
 SELECT @@global.innodb_default_encryption_key_id;
+
+SET innodb_default_encryption_key_id=0;
+SELECT @@session.innodb_default_encryption_key_id;
+SELECT @@global.innodb_default_encryption_key_id;
+
+SET global innodb_default_encryption_key_id=0;
+SELECT @@session.innodb_default_encryption_key_id;
+SELECT @@global.innodb_default_encryption_key_id;
+
```

Comment:
can you set to non-zero before SET PERSIST? that way we will know whether SET PERSIST changed the value or not?

Reply:
Same for PERSIST_ONLY